### PR TITLE
Add to DeBERTa resources

### PR DESCRIPTION
**Pull request by Saad135**
# What does this PR do?

Adds resources to Deberta

Relates to https://github.com/huggingface/transformers/issues/20055

@stevhliu Can you please take a look :-). I could not really find anything on DeBERTa but since DeBERTa builds upon RoBERTa, should I add the materials for RoBERTa?

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: docs/source/en/model_doc/deberta.mdx [hunk 1]
```diff
@@ -38,6 +38,13 @@ pre-trained models will be made publicly available at https://github.com/microso
 This model was contributed by [DeBERTa](https://huggingface.co/DeBERTa). This model TF 2.0 implementation was
 contributed by [kamalkraj](https://huggingface.co/kamalkraj) . The original code can be found [here](https://github.com/microsoft/DeBERTa).
 
+## Resources
+
+A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with DeBERTa. If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
+
+<PipelineTag pipeline="token-classification" />
+
+- [Byte-Pair Encoding tokenization](https://huggingface.co/course/chapter7/2?fw=pt) chapter of the 🤗 Hugging Face Course.
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta.mdx:47]:
```suggestion
<PipelineTag pipeline="token-classification"/>

- [Byte-Pair Encoding tokenization](https://huggingface.co/course/chapter6/5?fw=pt) chapter of the 🤗 Hugging Face Course.
```

**Review by stevhliu — COMMENTED** [review]
Thanks for the contribution! 🤗 

I think we should keep the materials specific to DeBERTa for now. There are still several resources you can add, for example, from the Hugging Face blog:

- [Accelerate Large Model Training Using DeepSpeed](https://huggingface.co/blog/accelerate-deepspeed)
- [Supercharged Customer Service with Machine Learning](https://huggingface.co/blog/supercharge-customer-service-with-machine-learning)

You can also add the example scripts and notebooks for the tasks DeBERTa supports: masked language modeling, sequence classification, token classification, and question answering.

File: docs/source/en/model_doc/deberta.mdx (hunk 1 above)
Comment (Saad135) [reply docs/source/en/model_doc/deberta.mdx]: Should be fixed now.

**Review by stevhliu — COMMENTED** [review]
Thanks, just one more minor change and we should be ready! :)

File: docs/source/en/model_doc/deberta.mdx [hunk 2]
```diff
@@ -38,6 +38,39 @@ pre-trained models will be made publicly available at https://github.com/microso
...
+- A blog post on [Supercharged Customer Service with Machine Learning](https://huggingface.co/blog/supercharge-customer-service-with-machine-learning) with DeBERTa.
+- [`DebertaForSequenceClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/text-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/text_classification.ipynb).
+- [`TFDebertaForSequenceClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/text-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/text_classification-tf.ipynb).
+- [`FlaxDebertaForSequenceClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/flax/text-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/text_classification_flax.ipynb).
```
[the remark this answers is not in the document]
Comment (Saad135) [reply docs/source/en/model_doc/deberta.mdx]: Done :-)

**Review by stevhliu — APPROVED** [review]
Very nice, looks good to me! Pinging @sgugger for a final look :)

**Review by sgugger — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**